pkg/expect/expect.go

```diff
@@ -223,14 +229,29 @@ func (ep *ExpectProcess) ExpectFunc(ctx context.Context, f func(string) bool) (s
 }
 
 // ExpectWithContext returns the first line containing the given string.
-func (ep *ExpectProcess) ExpectWithContext(ctx context.Context, s string) (string, error) {
-	return ep.ExpectFunc(ctx, func(txt string) bool { return strings.Contains(txt, s) })
+func (ep *ExpectProcess) ExpectWithContext(ctx context.Context, s ExpectedResponse) (string, error) {
+	var (
+		expr *regexp.Regexp
+		err  error
+	)
+	if s.IsRegularExpr {
+		expr, err = regexp.Compile(s.Value)
+		if err != nil {
+			return "", err
+		}
+	}
+	return ep.ExpectFunc(ctx, func(txt string) bool {
+		if expr != nil {
+			return expr.MatchString(txt)
+		}
+		return strings.Contains(txt, s.Value)
+	})
 }
 
 // Expect returns the first line containing the given string.
 // Deprecated: please use ExpectWithContext instead.
```

Comment:
This function is deprecated? Do we need to move it? Just want to mark, no need to remove in this PR,

Reply:
We can cleanup the code in separate PR.

Reply:
sgtm.